indico/web/client/js/custom_elements/ind_menu.js

```diff
@@ -0,0 +1,56 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2023 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import './ind_menu.scss';
+
+let lastId = 0; // Track the assigned IDs to give each element a unique ID
+
+customElements.define(
+  'ind-menu',
+  class extends HTMLElement {
+    connectedCallback() {
+      window.addEventListener('DOMContentLoaded', () => {
```

Comment:
Is this still needed in a custom element? (e.g. if/because the `connectedCallback` runs before the element and its children have been fully loaded into he DOM)

Reply:
"Is this still needed in a custom element?"

Unfortunately yes. We'll want to do something about deferring all <script> tags so we don't have to do this everywhere. Luckily, it's not hard to remove later either.